src/poly/commitment.rs

```diff
@@ -21,6 +23,95 @@ pub struct OpeningProof<C: CurveAffine> {
     z2: C::Scalar,
 }
 
+/// A multiscalar multiplication in the polynomial commitment scheme
+#[derive(Debug)]
+pub struct MSM<C: CurveAffine> {
+    /// TODO: documentation
+    pub g_scalars: Option<Vec<C::Scalar>>,
+
+    /// TODO: documentation
+    pub h_scalar: Option<C::Scalar>,
+
+    /// TODO: documentation
+    pub other_scalars: Vec<C::Scalar>,
+
+    /// TODO: documentation
+    pub other_bases: Vec<C>,
+}
+
+impl<'a, C: CurveAffine> MSM<C> {
+    /// Empty MSM
+    pub fn default(params: &Params<C>) -> Self {
+        let g_scalars = Some(vec![C::Scalar::one(); params.n as usize]);
+        let h_scalar = Some(C::Scalar::one());
+        let other_scalars: Vec<C::Scalar> = Vec::with_capacity(params.k as usize * 2 + 3);
+        let other_bases: Vec<C> = Vec::with_capacity(params.k as usize * 2 + 3);
```

Comment:
I think that in practice because the MSM will be (eventually) passed into the plonk verifier, these initial capacities will be lower. This is typically a small vector anyway so we can just initialize these to `vec![]`.